CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+cmake_minimum_required(VERSION 3.1.0)
+project(fast_cpp_csv_parser)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+#------------------- COMPILATION FLAGS ----------------------------------------
+
+if (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "")
+    string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE)
+endif()
+
+if ("${CMAKE_BUILD_TYPE}" STREQUAL "RELEASE")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -O3")
+else()
+    set(CMAKE_BUILD_TYPE DEBUG)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -Wall -O0")
+endif()
+message(STATUS "Building ${CMAKE_PROJECT_NAME} in ${CMAKE_BUILD_TYPE} mode")
+
+#------------------- FILE IDENTIFICATION --------------------------------------
+
+set(FAST_CPP_CSV_PARSER_LIB ${CMAKE_PROJECT_NAME})
+set(FAST_CPP_CSV_PARSER_SRC
+    csv.h
+)
+
+#------------------- LIBRARY DECLARATION --------------------------------------
+
+add_library(${FAST_CPP_CSV_PARSER_LIB} ${FAST_CPP_CSV_PARSER_SRC})
+set_target_properties(${FAST_CPP_CSV_PARSER_LIB} PROPERTIES
+    LINKER_LANGUAGE CXX
+)
+#------------------- LIBRARY DEPENDENCY LINKING -------------------------------
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
```

Comment:
Currently there is a parameter `CSV_IO_NO_THREAD` to control if threads are used.  I would suggest wrapping this to maintain this behavior.

```
option(USE_THREADS "Enable the use of threads within fast-cpp-parser" ON)
if (USE_THREADS)
  set(THREADS_PREFER_PTHREAD_FLAG ON)
  find_package(Threads REQUIRED)
else
  add_definitions(-DCSV_IO_NO_THREAD)
endif()
```

Reply:
Agreed. Modifying the pull requests with your suggestions.